### ISSUE TITLE
Orchestration Frameworks

### ISSUE THREAD

**Issue by GoranPaues** [KTH/devops-course#6]
Once you have several containerized modules, the next problem is to specify how they should work together and how they should be operated. Here the orchestration frameworks like Kubernetes and Docker Swarm play an important role. From my perpective it looks like Kubernetes has a big momentum here. Google Cloud has had a Kubernetes service for years and the other cloud providers are hard at work producing similar solutions.

**Comment by SimoneStefani** [comment]
I agree with @GoranPaues that **Kubernetes** is the way to go. Google launches ~4 billion containers every week for their internal infrastructure so they know very well the job...

Probably is possible to contact even Google itself for some training. Tricky thing with orchestration is that it's a very high layer which requires a lot of knowledge on the lower levels.

**Comment by MatsJonsson** [comment]
+1 for Kubernetes. We recommend Kelsey Hightower´s excellent introduction "Kubernetes the Hard Way", available on gihub https://github.com/kelseyhightower/kubernetes-the-hard-way

**Comment by bbaudry** [comment]
https://www.linux.com/NEWS/8-OPEN-SOURCE-CONTAINER-ORCHESTRATION-TOOLS-KNOW

**Comment by monperrus** [comment]
kata container
https://katacontainers.io/

**Comment by monperrus** [comment]
Containerd
https://containerd.io/

**Comment by monperrus** [comment]
rkt, a security-minded, standards-based container engine
https://coreos.com/rkt/

**Comment by monperrus** [comment]
Docker Swarm
https://docs.docker.com/engine/swarm/

**Comment by monperrus** [comment]
Easily Deploy Applications with Nomad
https://www.nomadproject.io/

**Comment by monperrus** [comment]
Concept of API Gateway
https://microservices.io/patterns/apigateway.html

**Comment by monperrus** [comment]
OpenNebula is a simple yet powerful and flexible turnkey open-source solution to build Private Clouds and manage Data Center virtualization.
https://opennebula.org/

**Comment by monperrus** [comment]
Deploying Your First Docker Container in the browser directly
https://www.katacoda.com/courses/docker/deploying-first-container

**Comment by bbaudry** [comment]
What is an operator?
https://youtu.be/HuVlz0lM7Yo

**Comment by monperrus** [comment]
Edge Router / Reverse proxy
https://github.com/containous/traefik

**Comment by monperrus** [comment]
Learning from, Understanding, and Supporting DevOps Artifacts for Docker
https://arxiv.org/pdf/2002.03064

**Comment by bbaudry** [comment]
[Tekton](https://cloud.google.com/tekton/)
Kubernetes-native open-source framework for creating continuous integration and delivery

**Comment by bbaudry** [comment]
Increment issue on containers

https://increment.com/containers/

**Comment by mrbgco** [comment]
[CloudNative landscape](https://landscape.cncf.io/?grouping=no).

**Comment by bbaudry** [comment]
The lifecycle of containers is an interesting topic
https://dev.to/alubhorta/the-docker-container-lifecycle-docker-made-easy-3-554o

**Comment by bbaudry** [comment]
Quarkus is a relevant technology for containers
https://quarkus.io/

**Comment by monperrus** [comment]
buildkit: concurrent, cache-efficient, and Dockerfile-agnostic builder toolkit
https://github.com/moby/buildkit

**Comment by monperrus** [comment]
Static Dockerfile checkers [Hadolint](https://github.com/hadolint/hadolint) or [Conftest](https://github.com/open-policy-agent/conftest)

**Comment by monperrus** [comment]
KEDA | Kubernetes Event-driven Autoscaling
https://keda.sh/

**Comment by monperrus** [comment]
Helm is the package manager for Kubernetes
https://helm.sh/

**Comment by mrbgco** [comment]
Another package manager for K8s
https://kustomize.io/

**Comment by monperrus** [comment]
DC/OS (the Distributed Cloud Operating System) is an open-source, distributed operating system based on the Apache Mesos distributed systems kernel.
https://dcos.io/

**Comment by bbaudry** [comment]
Gtihub codespaces
https://github.com/features/codespaces

**Comment by bbaudry** [comment]
fresh, automated dev environments for each task
https://www.gitpod.io/

**Comment by monperrus** [comment]
[Visual notations in container orchestrations: an empirical study with Docker Compose](http://link.springer.com/10.1007/s10270-022-01027-8)

**Comment by monperrus** [comment]
[Modus: a Datalog dialect for building container images.](https://dblp.org/rec/conf/sigsoft/TomyWBM22)

**Comment by bbaudry** [comment]
a tool for deploying to NixOS machines in a network or cloud
https://github.com/NixOS/nixops

**Comment by monperrus** [comment]
Deploy, configure, troubleshoot and secure containers in minutes on Kubernetes, Docker, and Swarm
https://www.portainer.io/
